From eb01f9e2d13acb843a383eb7c36b7eedfbcb2d1f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:46 -0400
Subject: [usb] wusbcore: use list_move_tail instead of list_del/list_add_tail

Message-id: <1410887463-20674-95-git-send-email-dzickus@redhat.com>
Patchwork-id: 95302
O-Subject: [RHEL7 PATCH 094/591] USB: WUSBCORE: use list_move_tail instead of list_del/list_add_tail
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 8eb4129929e9eb563fbab807a2b274d32042d034
Author: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date:   Mon Sep 23 14:16:22 2013 +0800

    USB: WUSBCORE: use list_move_tail instead of list_del/list_add_tail

    Using list_move_tail() instead of list_del() + list_add_tail().

    Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-xfer.c b/drivers/usb/wusbcore/wa-xfer.c
index 6ad02f5..cf10979 100644
--- a/drivers/usb/wusbcore/wa-xfer.c
+++ b/drivers/usb/wusbcore/wa-xfer.c
@@ -1552,10 +1552,8 @@ error_complete:
 
   dev_info(dev, "Control EP stall.  Queue delayed work.\n");
   spin_lock_irq(&wa->xfer_list_lock);
-  /* remove xfer from xfer_list. */
-  list_del(&xfer->list_node);
-  /* add xfer to xfer_errored_list. */
-  list_add_tail(&xfer->list_node, &wa->xfer_errored_list);
+  /* move xfer from xfer_list to xfer_errored_list. */
+  list_move_tail(&xfer->list_node, &wa->xfer_errored_list);
   spin_unlock_irq(&wa->xfer_list_lock);
   spin_unlock_irqrestore(&xfer->lock, flags);
   queue_work(wusbd, &wa->xfer_error_work);
-- 
1.7.1